packages/mainsail/source/transaction.service.ts

```diff
@@ -63,171 +60,91 @@ export class TransactionService extends Services.AbstractTransactionService {
 				`[TransactionService#transfer] Expected fee to be defined but received ${typeof input.fee}`,
 			);
 		}
+	}
+
+	public override async transfer(input: Services.TransferInput): Promise<Contracts.SignedTransactionData> {
+		applyCryptoConfiguration(this.#configCrypto);
+		this.#validateInput(input);
+
+		const transaction = this.#app.resolve(EvmCallBuilder);
 
-		return this.#createFromData("transfer", input, ({ transaction, data }) => {
-			transaction.recipientId(data.to);
+		const { address } = await this.#signerData(input);
+		const nonce = await this.#generateNonce(address, input);
 
-			if (data.memo) {
-				transaction.vendorField(data.memo);
-			}
-		});
+		transaction
+			.network(Number(this.#configCrypto.crypto.network.nethash))
+			.gasLimit(GasLimit.Transfer)
+			.recipientAddress(input.data.to)
+			.payload("")
+			.nonce(nonce)
+			.value(this.toSatoshi(input.data.amount).toString()) // revisit
+			.gasPrice(this.toSatoshi(input.fee).toNumber()); // revisit
```

Comment:
better to use the `formatUnits` and `parseUnits` methods now as those will use the right values. Amount will be in `wei`, and fee will be in `gwei`. For fee it's likely that you will use the value as it gets passed (e.g. `5` for a transfer, will be `5 gwei`) while for the amount you want it in wei while it gets passed as ark (I believe), so you'd want to write the ark value in gwei instead = use `parseUnits(input.data.amount, 'ark')`